03 - Clustering Misorientations.ipynb

```diff
@@ -165,120 +246,296 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from orix.vector import Vector3d\n",
+    "from orix.quaternion.rotation import Rotation\n",
+    "from orix.vector.neo_euler import AxAngle"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From Krakow et al\n",
+    "sigma7a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(64.40)))\n",
+    "sigma11a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(34.96)))\n",
+    "sigma11b = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(85.03)))\n",
+    "sigma13a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(76.89)))\n",
+    "sigma13b = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(57.22)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From Laine et al\n",
+    "sigma7a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(64.40)))\n",
+    "sigma11a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(34.96)))\n",
+    "sigma11b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 1, 0), np.radians(85.03)))\n",
+    "sigma13a = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(76.89)))\n",
+    "sigma13b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 1, 0), np.radians(57.22)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# From Bonnet et al\n",
+    "sigma7a = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(64.62)))\n",
+    "sigma11a = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(35.10)))\n",
+    "sigma11b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(84.78)))\n",
+    "sigma13a = Rotation.from_neo_euler(AxAngle.from_axes_angles((2, 1, 0), np.radians(76.66)))\n",
+    "sigma13b = Rotation.from_neo_euler(AxAngle.from_axes_angles((1, 0, 0), np.radians(57.42)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "twin_theory = Orientation.stack([sigma7a, sigma11a, sigma11b, sigma13a, sigma13b]).flatten()"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "<h2><center> Plotting our results </h2></center>"
+    "Calculate difference, defined as minimum rotation angle, between measured and theoretical values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "difference = Misorientation((~cluster_means).outer(twin_theory)).set_symmetry(D6, D6).angle.data"
```

Comment:
I am fairly sure that this isn't guaranteed to work.

Reply:
Yeah I agree with that, hence the second go with symmetry